### PR TITLE
Adding Java Sample with getting Request-Id

### PR DESCRIPTION
**Pull request by TsuyoshiUshio**
## Purpose

This PR provide the Java Sample template for Application Insights. I'm using Spring Boot with Gradle. 

Currently, I check it works with Azure Functions. It receive http request with correlation telemetry. 
For more detail, I wrote README. Please refer it. 

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ x ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ x ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test

Read the README.md

## What to Check

I'm not good at Java development. I'm not sure if my code fit the regulation of the Java development. 
If you don't like the naming of the class, feel free to refactor it.

### CODE REVIEW

File: JavaSample/build.gradle [hunk 1]
```diff
@@ -28,4 +30,18 @@ targetCompatibility = 1.8
...
\ No newline at end of file
+    compile group: 'com.microsoft.azure', name: 'applicationinsights-web', version: '2.+'
+}
+
+sourceSets.main.resources {
+    srcDirs = ['src/main/java']
+    include '**/*.xml'
+}
+
+// task copy(type:Copy) {
```
Comment (akirakakar) [inline JavaSample/build.gradle]: Do we need to keep this comment?

File: JavaSample/src/main/java/processor/configurations/AppInsightsConfig.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+     * @return {@link Bean} of type {@link WebRequestTrackingFilter}
+     */
+    @Bean
+    @ConditionalOnMissingBean
+
+    public WebRequestTrackingFilter webRequestTrackingFilter(@Value("${spring.application.name:application}") String applicationName) {
+        return new WebRequestTrackingFilter(applicationName);
+    }
+
+
```
Comment (akirakakar) [inline JavaSample/src/main/java/processor/configurations/AppInsightsConfig.java]: Sorry to be really picky here
nit: Spacing does not seem to be consistent.

File: JavaSample/src/main/java/processor/configurations/AppInsightsConfig.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+import com.microsoft.applicationinsights.TelemetryConfiguration;
+import com.microsoft.applicationinsights.web.internal.WebRequestTrackingFilter;
+
+@Configuration
+public class AppInsightsConfig {
+
+    @Bean
+    public String telemetryConfig() {
+        String telemetryKey = System.getenv("APPINSIGHTS_INSTRUMENTATIONKEY");
+        if (telemetryKey != null) {
```
Comment (akirakakar) [inline JavaSample/src/main/java/processor/configurations/AppInsightsConfig.java]: Shouldn't you do empty check as well here?
https://google.github.io/guava/releases/21.0/api/docs/com/google/common/base/Strings.html#isNullOrEmpty-java.lang.String-

File: JavaSample/src/main/java/processor/configurations/AppInsightsConfig.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+
+    /**
+     * Creates bean of type WebRequestTrackingFilter for request tracking
+     * @param applicationName Name of the application to bind filter to
+     * @return {@link Bean} of type {@link WebRequestTrackingFilter}
+     */
+    @Bean
+    @ConditionalOnMissingBean
+
```
Comment (akirakakar) [inline JavaSample/src/main/java/processor/configurations/AppInsightsConfig.java]: nit: space.

File: JavaSample/src/main/java/processor/Message.java [hunk 5]
```diff
@@ -0,0 +1,13 @@
+package processor;
+
+public class Message {
+    private final String text;
+
+    public Message(String text) {
```
Comment (akirakakar) [inline JavaSample/src/main/java/processor/Message.java]: Just a suggestion but maybe use getter and constructor from lombok? https://projectlombok.org/features/GetterSetter

**Review by akirakakar — COMMENTED** [review]
Do we need to commit .vscode/ ? 
Also, adding lombok is nice but not required since application is not big.

**Comment by TsuyoshiUshio** [comment]
Done for removing .vscode

File: JavaSample/build.gradle (hunk 1 above)
Comment (TsuyoshiUshio) [reply JavaSample/build.gradle]: Done

File: JavaSample/src/main/java/processor/configurations/AppInsightsConfig.java (hunk 2 above)
Comment (TsuyoshiUshio) [reply JavaSample/src/main/java/processor/configurations/AppInsightsConfig.java]: Done. Do you know a good tool for checking the style?

File: JavaSample/src/main/java/processor/configurations/AppInsightsConfig.java (hunk 3 above)
Comment (TsuyoshiUshio) [reply JavaSample/src/main/java/processor/configurations/AppInsightsConfig.java]: Done as well. I also add the Google Core Libraries.

File: JavaSample/src/main/java/processor/configurations/AppInsightsConfig.java (hunk 4 above)
Comment (TsuyoshiUshio) [reply JavaSample/src/main/java/processor/configurations/AppInsightsConfig.java]: Done.

File: JavaSample/src/main/java/processor/Message.java (hunk 5 above)
Comment (TsuyoshiUshio) [reply JavaSample/src/main/java/processor/Message.java]: I use Lombok as the first time. However, I encounter the issue to configure the lombok of gradle ver 5.x. However, I solved this now.

**Comment by TsuyoshiUshio** [comment]
@akirakakar Could your review it again? if it is ok, please merge it.

**Review by akirakakar — APPROVED** [review]
Thanks. lgtm

**Pull request merged**